src/Tasks/Microsoft.NET.Build.Tasks/WriteAppConfig.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public sealed class WriteAppConfig : TaskBase
+    {
+        /// <summary>
+        /// Path to the app.config source file.
+        /// </summary>
+        public ITaskItem AppConfigFile { get; set; }
+
+        [Required]
+        public string TargetFrameworkIdentifier { get; set; }
+
+        [Required]
+        public string TargetFrameworkVersion { get; set; }
+
```

Comment:
For completeness, this should have optional TargetFrameworkProfile as well.

Reply:
Added TargetFrameworkProfile